client/my-sites/earn/home.tsx

```diff
@@ -239,11 +238,10 @@ const Home: FunctionComponent< ConnectedProps > = ( {
 export default connect< ConnectedProps, {}, {} >( state => {
 	const selectedSiteSlug = getSelectedSiteSlug( state );
 	const site = getSiteBySlug( state, selectedSiteSlug );
-	const plan = getCurrentPlan( state, site.ID );
 	return {
 		siteId: site.ID,
 		selectedSiteSlug,
-		isFreePlan: get( plan, 'productSlug', '' ) === PLAN_FREE,
+		isFreePlan: ! isCurrentPlanPaid( state, site.ID ),
```

Comment:
FYI: There's `isSiteOnFreePlan`

https://github.com/Automattic/wp-calypso/blob/573e9e91b22ba36c4c4797b4bb13d4cb559b00a0/client/state/selectors/is-site-on-free-plan.js#L20